src/tbb/waiters.h

```diff
@@ -68,6 +68,7 @@ class outermost_worker_waiter : public waiter_base {
                     t2 = std::chrono::steady_clock::now())
                 {
                     if (!my_arena.is_empty() && !my_arena.is_recall_requested()) {
+                        reset_wait();
```

Comment:
Can you please add a comment in the code as well, why the reset is required

Reply:
Agreed to move this change to a separate patch.